### PR TITLE
Adjustments to split `FunctionParameterSyntax` into multiple nodes for function parameters, closure parameters and enum parameters

### PR DESCRIPTION
**Pull request by ahoppen**
Companion of https://github.com/apple/swift-syntax/pull/1455

### CODE REVIEW

File: Sources/SwiftFormatPrettyPrint/TokenStreamCreator.swift [hunk 1]
```diff
@@ -2606,6 +2606,24 @@ fileprivate final class TokenStreamCreator: SyntaxVisitor {
...
   }
 
+  /// Applies formatting to a collection of enum case parameters for a decl.
+  ///
+  /// - Parameters:
+  ///    - parameters: A node that contains the parameters that can be passed to a decl when its
+  ///      called.
+  ///    - forcesBreakBeforeRightParen: Whether a break should be required before the right paren
+  ///      when the right paren is on a different line than the corresponding left paren.
+  private func arrangeEnumCaseParameterClause(
```
Comment (ahoppen) [inline Sources/SwiftFormatPrettyPrint/TokenStreamCreator.swift:2616]: I just duplicated this method for `EnumCaseParameterClauseSyntax`. Alternatives would be
1. Make this function take a `leftParen: TokenSyntax, isEmpty: Bool, rightParen: TokenSyntax` so the logic can be shared with `arrangeParameterClause`
2. Create a protocol that offers the three parameters described in (1) and make both `ParameterClauseSyntax` and `EnumCaseParameterClauseSyntax` conform to that

(2) seemed like quite an overkill to me and (1) seemed ugly as well. Let me know if you’ve got preferences @allevato.

Comment (allevato) [reply Sources/SwiftFormatPrettyPrint/TokenStreamCreator.swift:2616]: This is fine; the function is small enough that duplicating it is probably better than overengineering a generic thing that would only be used in one extra place.

**Review by allevato — APPROVED** [review]

File: Sources/SwiftFormatPrettyPrint/TokenStreamCreator.swift (hunk 1 above)
Comment (ahoppen) [reply Sources/SwiftFormatPrettyPrint/TokenStreamCreator.swift:2616]: That’s what I thought as well. But good to hear you’re on the same page.

**Comment by ahoppen** [comment]
@allevato I have needed to duplicate some more code to make sure closure and enum case parameters are correctly formatted. Could you take another look whether you think it makes sense to unify some of these methods? I still think that it might be too much overhead but you decide.

**Comment by allevato** [comment]
I still think the duplication is small enough that it's fine to have it for now, especially since all of these param lists are subtly different. If I want to revisit it in the future I can do that as part of a larger refactoring, but no reason to block you on it.

**Pull request merged**